### PR TITLE
Bluetooth: BTP: Extend BTP with Mesh related commands 

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This adds BTP commands needed to test Mesh against PTS

### CODE REVIEW

File: tests/bluetooth/tester/btp_spec.txt [hunk 1]
```diff
@@ -1391,6 +1391,44 @@ Commands and responses:
...
+		while in IV Update test mode, ignoring the 96 hour limit.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x0a - Network Send
+		Controller Index:	<controller id>
+		Command parameters:	Valid parameters (1 octet)
+					TTL (1 octet)
+					SRC (2 octets)
+					DST (2 octets)
```
Comment (jhedberg) [inline tests/bluetooth/tester/btp_spec.txt]: Isn't there the actual payload parameter missing here?

File: tests/bluetooth/tester/btp_spec.txt [hunk 2]
```diff
@@ -1391,6 +1391,44 @@ Commands and responses:
...
+					0x01 = IV Update in Progress
+
+		This command is used to make a IV Update state transition
+		while in IV Update test mode, ignoring the 96 hour limit.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x0a - Network Send
+		Controller Index:	<controller id>
+		Command parameters:	Valid parameters (1 octet)
```
Comment (jhedberg) [inline tests/bluetooth/tester/btp_spec.txt]: What does "Valid parameters (1 octet)" mean?

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: tests/bluetooth/tester/btp_spec.txt (hunk 1 above)
Comment (MariuszSkamra) [reply tests/bluetooth/tester/btp_spec.txt]: I'll explain myself ;) I thought about adding this, but I saw that PTS is not for any particular data to be sent, so I just skipped this and I wanted to add payload in handler of this command.
But on the other hand this command would be more versatile when payload will be here.

File: tests/bluetooth/tester/btp_spec.txt (hunk 2 above)
Comment (MariuszSkamra) [reply tests/bluetooth/tester/btp_spec.txt]: TTL parameter is not always given in PTS MMI, so I needed to some bitfield to indicate which paramaters has to be taken into account

**Comment by MariuszSkamra** [comment]
updated

**Comment by MariuszSkamra** [comment]
few mode commands & events added

**Comment by MariuszSkamra** [comment]
Health Generate Faults & Health Clear Faults commands added

**Comment by MariuszSkamra** [comment]
rebased

**Review by sjanc — APPROVED** [review]

**Comment by MariuszSkamra** [comment]
rebased

File: tests/bluetooth/tester/btp_spec.txt [hunk 3]
```diff
@@ -1510,3 +1569,15 @@ Events:
...
 		This event indicates that provisioning link has been closed on
 		given bearer.
+
+	Opcode 0x86 - Network receive Event
+
+		Controller Index:	<controller id>
+		Event parameters:	TTL (1 octet)
+					CTL (1 octet)
+					SRC (2 octets)
+					DST (2 octets)
```
Comment (jhedberg) [inline tests/bluetooth/tester/btp_spec.txt]: The chosen information seems a bit arbitrary to me. So we don't need information on which IVI was used, which NetKey Index was used, etc? I suppose the payload is the payload after decryption with the network key? Don't some tests also require verification of data after app key decryption?

Comment (MariuszSkamra) [reply tests/bluetooth/tester/btp_spec.txt]: Of course, we can extend the event information if it'll be needed. I choose those, to satisfy PTS needs.
````
Please confirm the following network packet was received: Network Header: 
    TTL: [0x00]
    CTL: [0x00]
    SRC: [0x0001]
    DST: [0xFFFC]
    TransportPDU: [0x01020304050607080910111213141516]
````
Yes, the payload here is the payload is after decryption.
I did not saw such tests yet.

Comment (jhedberg) [reply tests/bluetooth/tester/btp_spec.txt]: After decryption with the NetKey or after decryption with NetKey+AppKey?

Comment (MariuszSkamra) [reply tests/bluetooth/tester/btp_spec.txt]: decryption with the netkey

**Review by jhedberg — APPROVED** [review]

**Pull request merged**